scripts/ci/run_perftest.sh

```diff
@@ -117,36 +151,73 @@ for scale in 50 100 500; do
     done
 
-    # delete db to start anew
-    delete_response=$(curl -s -o /dev/null -X DELETE -H "Authorization: Token $TEST_TOKEN" "http://${NGINX_HOST}:8086/api/v2/buckets/$bucket_id?org=$TEST_ORG" -w %{http_code})
-    if [ $delete_response != "204" ]; then
-      echo "bucket not deleted!"
-      exit 1
-    fi
+    clean_db
   done
 done
 
+# Metaquery tests
+
+# Controls how many data points will be generated. The total number of points
+# will be scale_var * scale_var, where each point is (approximately) a unique
+# series.
+scale_var=1000
+
+# How many queries to generate.
+queries=100
+
+# The time range controls both the span over which data is generated, and the
+# span over which queries will be performed. timestamp-start and timestamp-end
+# must be provided to both the data generation and query generation commands
+# and must be the same to ensure that the queries cover the data range.
+start="2019-01-01T00:00:00Z"
+end="2020-01-01T00:00:00Z"
+
+# How long to run the query phase of the benchmarks. It's best to specify a
+# duration to limit the maximum amount of time the queries can run for these,
+# since individual queries can take a long time to run.
+duration=30s
+
+# Generate data
+data_fname="influx-bulk-records-usecase-metaquery.txt"
+$GOPATH/bin/bulk_data_gen --seed=$seed -use-case metaquery -scale-var $scale_var -timestamp-start $start -timestamp-end $end > ${DATASET_DIR}/$data_fname
+
+# Generate flux queries for field-keys and tag-values
+query_files=""
+for type in field-keys tag-values; do
+  for format in http flux-http; do
+    query_fname="query-metaquery-$type-$format.txt"
+    $GOPATH/bin/bulk_query_gen -use-case metaquery -query-type $type -format influx-$format -timestamp-start $start -timestamp-end $end -queries $queries > ${DATASET_DIR}/$query_fname
+    query_files="$query_files $query_fname"
+  done
+done
+
+# Load the data
+load_opts="-urls=http://${NGINX_HOST}:8086 -do-abort-on-exist=false -do-db-create=true"
+if [ -z $INFLUXDB2 ] || [ $INFLUXDB2 = true ]; then
+  load_opts="$load_opts -organization=$TEST_ORG -token=$TEST_TOKEN"
+fi
+cat ${DATASET_DIR}/$data_fname | $GOPATH/bin/bulk_load_influx $load_opts > /dev/null
+
+force_compaction
+
+# Run the query benchmarks
+for query_file in $query_files; do
+  format=$(echo $query_file | cut -d '-' -f3,4,5)
+  format=${format%.txt}
+  cat ${DATASET_DIR}/$query_file | ${GOPATH}/bin/query_benchmarker_influxdb --urls=http://${NGINX_HOST}:8086 --benchmark-duration=$duration --debug=0 --print-interval=0 --json=true --organization=$TEST_ORG --token=$TEST_TOKEN | jq ". += {branch: \"${INFLUXDB_VERSION}\", commit: \"${TEST_COMMIT}\", time: \"$datestring\", i_type: \"${DATA_I_TYPE}\", query_format: \"$format\"}" > $working_dir/test-query-metaquery-format-$format.json
+done
+
+# clean up the DB before exiting
+clean_db
+
 echo "Using Telegraph to report results from the following files:"
```

Comment:
Depending on how separated-out you want these results to be from the general query tests, it may help to add a tag to the Telegraph config to indicate that these are metaquery tests for easy parsing of results, e.g. something like:

```
  tag_keys = [
    "i_type",
    "query_format",
    "is_metaquery"
  ]
  ```
  
  And then in the piping of the results into jq prior to writing them to a file something like:
  ```
|  jq ". += {is_metaquery: true, branch: ... }" ...
```